node/metrics/useQuote.ts

```diff
@@ -45,9 +55,7 @@ const buildUseQuoteMetric = (
       user_email: userEmail,
       quote_last_update: quote.lastUpdate,
     },
-  }
-
-  return metric
+  } as UseQuoteMetric
```

Comment:
You had to do this cast because you made `UseQuoteMetric` into a class, so Typescript won't recognize an object created directly like this one as having that type, only objects created with `new UseQuoteMetric`. They're not 100% equivalent in the end, so Typescript is right. Better to make `UseQuoteMetric`  a regular type or interface like I suggested further above.

Reply:
I tested with previously code, and with 'type', the tslint had some errors and there are others problems that I commented.
![image](https://github.com/vtex-apps/b2b-quotes-graphql/assets/5332361/40e189e0-d623-4051-a7ff-bc8e13b7e9fa)

Reply:
That's just asking you to use `interface` instead of `type`. So you'd do `interface UseQuoteFieldsMetric {` instead of `type UseQuoteFieldsMetric = {`. Note that `interface` in Typescript is just a way of defining an object type, it's nothing like interfaces in Java or in other languages.

Reply:
`type` and `interface` are almost the same thing in Typescript, but `interface` allows extending from other interfaces, which helps abstract types by their superclass names and avoiding repetition. So it's usually preferred, unless you want to do some operations that only work with `type` (such as `&`).

Reply:
@mairatma Is it ok use class here, as [the commented](https://github.com/vtex-apps/b2b-quotes-graphql/pull/42#discussion_r1379208925)?

Reply:
@Rudge it works of course, and it's ok if you insist. But it feels like you're writing in a different language, it's weird for other devs reading it. Dropping the class and keeping the duplication when creating the object directly, as the code was already doing, would be more in sync with the usual JS way of coding. Classes are generally only used when you really get the benefit of having instances due to complex calculations within methods, but they're never used just to hold data like this (this is the first time I've seen it actually hehe).

It's a language thing, I know that other languages heavily use classes, but JS is the opposite. The class syntax exists, but it's very new and it's actually not implemented in the same way behind the scenes. So you'll notice that JS codebases tend to be almost functional.